1-bootstrap/README.md

```diff
@@ -1,32 +1,381 @@
+
+The same pool can be used across multiple steps. Reserving a wider IP range allows for more concurrent builds. A /24 range supports 254 hosts.
+
+- __IAM Roles:__ The identity deploying the module must have the following IAM roles on the Private Worker Pool project:
+
+   - __Cloud Build WorkerPool User:__ `roles/cloudbuild.workerPoolUser`
+   - __Project IAM Admin:__ `roles/resourcemanager.projectIamAdmin`
+
+   ```bash
+   gcloud projects add-iam-policy-binding YOUR_WORKER_POOL_PROJECT_ID \
+   --member="serviceAccount:YOUR_SERVICE_ACCOUNT_EMAIL" \
+   --role="roles/cloudbuild.workerPoolUser"
+
+   gcloud projects add-iam-policy-binding YOUR_WORKER_POOL_PROJECT_ID \
+   --member="serviceAccount:YOUR_SERVICE_ACCOUNT_EMAIL" \
+   --role="roles/resourcemanager.projectIamAdmin"
+   ```
+
+
+### Environments Infrastructure
+
+The following infrastructure is required for the different environments:
+
+#### Folders
+
+You need to pre-create the following folders:
+
+- A common folder.
+- A folder per environment (e.g., development, nonproduction, production).
+- __IAM Roles:__ The identity deploying the module requires the following IAM roles on the folders:
+   - Folder Admin: `roles/resourcemanager.folderAdmin`
+   - Project Creator: `roles/resourcemanager.projectCreator`
+   - Compute Network Admin: `roles/compute.networkAdmin`
+   - Compute Shared VPC Admin: `roles/compute.xpnAdmin`
+
+   ```bash
+   gcloud resource-manager folders add-iam-policy-binding YOUR_FOLDER_ID \
+   --member="serviceAccount:YOUR_SERVICE_ACCOUNT_EMAIL" \
+   --role="roles/resourcemanager.folderAdmin"
+
+   gcloud resource-manager folders add-iam-policy-binding YOUR_FOLDER_ID \
+   --member="serviceAccount:YOUR_SERVICE_ACCOUNT_EMAIL" \
+   --role="roles/resourcemanager.projectCreator"
+
+   gcloud resource-manager folders add-iam-policy-binding YOUR_FOLDER_ID \
+   --member="serviceAccount:YOUR_SERVICE_ACCOUNT_EMAIL" \
+   --role="roles/compute.networkAdmin"
+
+   gcloud resource-manager folders add-iam-policy-binding YOUR_FOLDER_ID \
+   --member="serviceAccount:YOUR_SERVICE_ACCOUNT_EMAIL" \
+   --role="roles/compute.xpnAdmin"
+   ```
+
+### Shared Network
+
+You need a Shared VPC per environment already created.
+
+The networks must meet the following requirements:
+
+- Two subnets in different regions.
+- Each subnet must have two secondary ranges with at least /18 range.
+- A Cloud Nat configured to reach extenal repositories.
+
+Access [Best practices for GKE networking](https://cloud.google.com/kubernetes-engine/docs/best-practices/networking) for more information.
+
+For a network configuration example, check the [Foundation Shared VPC](https://github.com/terraform-google-modules/terraform-example-foundation/tree/main/4-appfactory) step.
```

Comment:
404 - page not found

Maybe it should be 

https://github.com/terraform-google-modules/terraform-example-foundation/tree/main/3-networks-svpc/modules/shared_vpc